src/Icons/src/Iconify.php

```diff
@@ -85,6 +85,52 @@ public function fetchIcon(string $prefix, string $name): Icon
         ]);
     }
 
+    public function fetchIcons(string $prefix, array $names): array
+    {
+        if (!isset($this->sets()[$prefix])) {
+            throw new IconNotFoundException(\sprintf('The icon set "%s" does not exist on iconify.design.', $prefix));
+        }
+
+        // Sort to enforce cache hits
+        $queryString = implode(',', $names);
+        if (!preg_match('#^[\w-]+(,[\w-]+)*$#', $queryString)) {
+            throw new \InvalidArgumentException('Invalid icon names.');
+        }
+
+        // 500 chars max
+        // 42 chars: https://api.iconify.design/mdi.json?icons=
+        if (450 < \strlen($queryString)) {
+            throw new \InvalidArgumentException('The query string is too long.');
```

Comment:
Please use dedicated exception classes

Reply:
Is the plan to catch this? (I typically only add named exceptions if this is the case - or it's realistic an end-user would want to catch)

Reply:
Nah I don't think so, but I like to have dedicated exceptions and also a base exception class per package (like they do on Symfony on packages)

Reply:
Not on every component to be far... ;) 

Yet I kinda agree to use custom exception but this should be done in a dedicated PR i think, carefully selecting the one we want to "migrate" and the one we do not. 

To explain myself here, this is internal and "should never" happen. 

But as we all know what "should never" means... i get your point.